packages/realm/src/Realm.ts

```diff
@@ -123,34 +122,27 @@ export class Realm {
 
   /**
    * Sets the log level for a specific category.
+   * @param options - The log options to use.
+   * @param options.level - The log level to be used by the logger. The default value is `info`.
+   * @param options.category - The category/component to set the log level for. If omitted, log level is set for all categories.
    * @note The log level can be changed during the lifetime of the application.
-   * @since 12.0.0
+   * @since 12.7.0
    * @example
    * Realm.setLogLevel({ category: LogCategory.Realm, level: "all" });
    */
   static setLogLevel(options: LogOptions): void;
 
   static setLogLevel(arg: LogLevel | LogOptions) {
-    const setLevel = (category: LogCategory, level: LogLevel) => {
-      assert(Object.values(LogCategory).includes(category));
-      const ref = binding.LogCategoryRef;
-      const categoryRef = ref.getCategory(category);
+    const setLevel = (level: LogLevel, category = LogCategory.Realm) => {
+      assert(Object.values(LogCategory).includes(category), `Unexpected log category: '${category}'`);
+      const categoryRef = binding.LogCategoryRef.getCategory(category);
       categoryRef.setDefaultLevelThreshold(toBindingLoggerLevel(level));
     };
 
     if (typeof arg === "string") {
-      for (const category of Object.values(LogCategory)) {
-        setLevel(category, arg);
-      }
+      setLevel(arg);
     } else {
-      const { level, category } = arg;
-      if (category) {
-        setLevel(category, level);
-      } else {
-        for (const category of Object.values(LogCategory)) {
-          setLevel(category, level);
-        }
-      }
+      setLevel(arg.level, arg.category);
```

Comment:
I love this simplification! 😄 I did look at that code and thought it must be possible to de-duplicate, but none presented itself. Great suggestion.